app/controllers/concerns/authenticable.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Authenticable
+  def current_user
+    return @current_user if @current_user
+
+    header = request.headers['Authorization']
+    return nil if header.nil?
+
+    decoded = JsonWebToken.decode(header)
+
+    @current_user = begin
+                      User.find(decoded[:user_id])
+    rescue StandardError
+                      raise ActiveRecord::RecordNotFound
+    end
```

Comment:
how about rescuing in the upper method current_user and instead of returning Not Found, (which means you are telling the client what users are in your DB) you return something like AuthenticationErrror, more generic.

Also: Is not quite good practice to raise ActiveRecord errors like that: If you catch the error to reraise it, then just leave it continue its course up in the backtrace (.find already throws an error which will try to be rescued from this place to the beginning of the backtrace)

Reply:
You could create the AuthenticationError here for now, like:
class AuthenticationError < StandardError end;

Later you can refactor to move it to the errors file in the following PR.

Reply:
Do you want some like this? 
```
    rescue Failure
        raise AuthenticationError
    end
  end

  class AuthenticationError < StandardError
  end
```
I don't know how to raise a new type of exception without a rescue

Reply:
Yes, rescue StandardError instead of Failure.